src/components/organisms/AdminVenueView/components/EventsView/EventsView.tsx

```diff
@@ -48,25 +56,50 @@ export const EventsView: React.FC<EventsViewProps> = ({ venueId, venue }) => {
 
   const renderedEvents = useMemo(
     () =>
-      events?.map((event) => {
-        return (
-          <TimingEvent
-            event={event}
+      events?.map((event) => (
+        <TimingEvent
+          event={event}
+          setShowCreateEventModal={setShowCreateEventModal}
+          setEditedEvent={setEditedEvent}
+          key={event.id}
+        />
+      )),
+    [events, setShowCreateEventModal, setEditedEvent]
+  );
+
+  const renderedSpaces = useMemo(() => {
+    const spaces = [...new Set(events?.map((event) => event.room))];
+    const getSpaceEvents = (space: string) =>
+      events?.filter((event) => event.room === space) ?? [];
+
+    return spaces?.map(
+      (space) =>
+        space && (
+          <TimingSpace
+            key={space}
+            spaceName={space}
+            spaceEvents={getSpaceEvents(space)}
             setShowCreateEventModal={setShowCreateEventModal}
             setEditedEvent={setEditedEvent}
-            key={event.id}
           />
-        );
-      }),
-    [events, setShowCreateEventModal, setEditedEvent]
-  );
+        )
+    );
+  }, [events, setShowCreateEventModal, setEditedEvent]);
 
   return (
     <>
       <div className="EventsView">
-        <h4 className="EventsView__title">Events Schedule</h4>
+        <div className="EventsView__header">
+          <h4 className="EventsView__title">Events Schedule</h4>
+          <Checkbox
+            checked={showSplittedEvents}
+            onChange={toggleSplittedEvents}
+            label="Split by Space"
```

Comment:
Can you disable the text selection for this title? Otherwise it's auto highlighted if you click twice on the toggle.

Reply:
fixed in 2afbf348cf9b4c04b752967251109f8b622ed0d6

Reply:
nice catch! fixed in 51bab9b6b7fb6ac5059fad492c271961b3e3e3f9